tools/compare.go

```diff
@@ -91,49 +164,119 @@ func compareBlocksE(cmd *cobra.Command, args []string) error {
+	}
+
+	isDiff, err := cmd.Flags().GetBool("diff")
+	if err != nil {
+		return fmt.Errorf("identifying --diff flag: %w", err)
+	}
+
+	if !isDiff {
+		for blockNum, blockExpected := range blocksExpected {
+			blocksCounted++
+			blockReceived, exists := blocksReceived[blockNum]
+			if !exists {
+				continue
+			}
+
+			if !proto.Equal(blockExpected, blockReceived) {
+				differentBlocks[blockExpected.Number] = block.Range{StartBlock: uint64(math.Round(float64(blockExpected.Number/100.0))) * 100,
+					ExclusiveEndBlock: (uint64(math.Round(float64(blockExpected.Number/100.0))) * 100) + 100}
+
+				if rangeIsGood {
+					rangeIsGood = false
+					fmt.Printf("bundle %d - %d is different\n", (rangeNum)*100000, (rangeNum+1)*100000)
+				}
+				fmt.Printf("Block #%v is different", blockExpected.Number)
+			}
+			if blocksCounted >= 100000 || uint64(blocksCounted) >= blockRange.Len()-uint64(rangeNum*100000) || blocksCounted == len(blocksExpected) {
+				if rangeIsGood {
+					fmt.Printf("✓ bundle %d - %d has no differences\n", (rangeNum)*100000, (rangeNum+1)*100000)
+				}
+				rangeIsGood = true
+				rangeNum++
+				blocksCounted = -1
+			}
 		}
+	} else {
+		if blockRange.Len() != 100 {
+			return fmt.Errorf("when using --diff, make size of range equal to 100")
+		}
+		for i := blockRange.StartBlock; i < blockRange.ExclusiveEndBlock; i++ {
+			blockReceived, exists := blocksReceived[i]
+			if !exists {
+				fmt.Printf("Block #%v is missing in %v\n", i, storeReceivedDef)
+			}
+			blockExpected, exists := blocksExpected[i]
+			if !exists {
+				fmt.Printf("Block #%v is missing in %v\n", i, storeExpectedDef)
+			}
 
-		blockAJSON, _ := json.Marshal(blockA)
-		blockBJSON, _ := json.Marshal(blockB)
+			if !proto.Equal(blockExpected, blockReceived) {
+				differentBlocks[blockExpected.Number] = block.Range{StartBlock: uint64(math.Round(float64(blockExpected.Number/100.0))) * 100,
+					ExclusiveEndBlock: (uint64(math.Round(float64(blockExpected.Number/100.0))) * 100) + 100}
 
-		diff := cmp.Diff(blockAJSON, blockBJSON)
-		if diff != "" {
-			fmt.Printf("❌ difference found on block id %s\n", blockID)
-			fmt.Println(diff)
-			return nil
+				fmt.Printf("Block #%v is different\n", blockExpected.Number)
+
+				blockExpectedJson, err := json.Marshal(blockExpected)
+				if err != nil {
+					return fmt.Errorf("marshaling block: %v", err)
+				}
+				blockReceivedJson, err := json.Marshal(blockReceived)
+				if err != nil {
+					return fmt.Errorf("marshaling block: %v", err)
+				}
+
+				diff := cmp.Diff(blockExpectedJson, blockReceivedJson)
+				if diff != "" {
+					fmt.Printf("%v\n\n", diff)
+				}
+			}
 		}
 	}
 
-	fmt.Println("✓ no differences found!")
+	fmt.Printf("\n\nTo see for details of the differences for the different bundles, run one of those commands:\n")
+	for _, blk := range differentBlocks {
+		fmt.Printf("- fireeth tools compare-blocks --diff=true %v %v %v-%v\n\n", storeExpectedDef, storeReceivedDef, blk.StartBlock, blk.ExclusiveEndBlock)
```

Comment:
It's rare we use `%v`, usually it's for debugging print a full object, important argument format:

- `"a bool with %t"`
- `"an integer with %d"`
- `"a string with %s"` (works with `string` or `fmt.Stringer` interface which struct implements via `func (o *Obj) String() string`)
- `"a float with 2 decimals %.2f"`
- `"the actual variable type with %T"`

Also, check object your are working with, often they have a `String() string` implementation so that `%s` is usuable directly